src/solvers/dgsem_p4est/dg_3d_parabolic.jl

```diff
@@ -0,0 +1,717 @@
+function prolong2boundaries!(cache_parabolic, flux_viscous,
+                             mesh::P4estMesh{3},
+                             equations_parabolic::AbstractEquationsParabolic,
+                             surface_integral, dg::DG, cache)
+    (; boundaries) = cache_parabolic
+    (; contravariant_vectors) = cache_parabolic.elements
+    index_range = eachnode(dg)
+
+    flux_viscous_x, flux_viscous_y, flux_viscous_z = flux_viscous
+
+    @threaded for boundary in eachboundary(dg, cache_parabolic)
+        # Copy solution data from the element using "delayed indexing" with
+        # a start value and a step size to get the correct face and orientation.
+        element = boundaries.neighbor_ids[boundary]
+        node_indices = boundaries.node_indices[boundary]
+        direction = indices2direction(node_indices)
+
+        i_node_start, i_node_step_i, i_node_step_j = index_to_start_step_3d(node_indices[1],
+                                                                            index_range)
+        j_node_start, j_node_step_i, j_node_step_j = index_to_start_step_3d(node_indices[2],
+                                                                            index_range)
+        k_node_start, k_node_step_i, k_node_step_j = index_to_start_step_3d(node_indices[3],
+                                                                            index_range)
+
+        i_node = i_node_start
+        j_node = j_node_start
+        k_node = k_node_start
+
+        for j in eachnode(dg)
+            for i in eachnode(dg)
+                # this is the outward normal direction on the primary element
+                normal_direction = get_normal_direction(direction, contravariant_vectors,
+                                                        i_node, j_node, k_node, element)
+
+                for v in eachvariable(equations_parabolic)
+                    flux_viscous = SVector(flux_viscous_x[v, i_node, j_node, k_node,
+                                                          element],
+                                           flux_viscous_y[v, i_node, j_node, k_node,
+                                                          element],
+                                           flux_viscous_z[v, i_node, j_node, k_node,
+                                                          element])
+
+                    boundaries.u[v, i, j, boundary] = dot(flux_viscous, normal_direction)
+                end
+                i_node += i_node_step_i
+                j_node += j_node_step_i
+                k_node += k_node_step_i
+            end
+            i_node += i_node_step_j
+            j_node += j_node_step_j
+            k_node += k_node_step_j
+        end
+    end
+    return nothing
+end
+
+# # Function barrier for type stability
+# !!! TODO: Figure out why this cannot removed eventhough it exists in the dg_2d_parabolic.jl file
```

Comment:
?

Reply:
Yes, we should resolve this. We noticed that several functions in `dg_3d_parabolic.jl` were duplicated in `dg_2d_parabolic.jl` as well, but removing them resulted in "function not found" specialization errors. We'll try to figure out why.